src/table/utils/handle-drag-columns.ts

```diff
@@ -0,0 +1,124 @@
+import React, { DragEvent } from 'react';
+import extend from 'extend';
+import { storeColumnWidths, updateColumnInfoOrders, fixTableHypercubeOrders } from './columns-sorting-util';
+import saveSoftProperties from './save-soft-properties';
+import { TableLayout } from '../../types';
+
+export interface Model extends EngineAPI.IGenericObject {
+  applyPatches: (arg0: Array<{ qPath: string; qOp: string; qValue: string }>, arg1: boolean) => Promise<void>;
+}
+
+type Column = {
+  width: number;
+  dataColIdx: number;
+};
+
+type DragStartProps = {
+  event: DragEvent;
+  layout: TableLayout;
+  cellRef: React.RefObject<HTMLTableElement>;
+  headRowRef: React.RefObject<HTMLTableElement>;
+  cell: Column;
+};
+
+type DragOverProps = {
+  event: DragEvent;
+  model: Model;
+  rtl: boolean;
+  columns: Column[];
+  setEngagedColumn: (i: number) => void;
+};
+
+type DragEndProps = {
+  event: DragEvent;
+  model: EngineAPI.IGenericObject;
+  setEngagedColumn: (arg0: undefined) => void;
+};
```

Comment:
I suggest moving those types into the types.ts file

Reply:
Sure